app/models/user.rb

```diff
@@ -1,8 +1,40 @@
 # frozen_string_literal: true
 
+require 'pdf-reader'
+
 class User < ActiveRecord::Base
   validates_presence_of :first_name, :last_name, :email
   validates_uniqueness_of :email
 
   has_many :residents, -> { order(position: :asc) }
+
+  def load_residents(file)
```

Comment:
I think this violates the Single Responsibility Principle, because now the User model has all these responsibilities related to parsing a file. It would be better to extract this code into a dedicated class.
I probably wouldn't even define a method on the User model. Once we have the `user_from_token` I would pass it to the class along with the file and let it do the work from there.

Reply:
Okay awesome! Yeah this did feel weird putting it in User. Makes alot more sense that way! Thanks!